src/lib/swapper/swappers/ThorchainSwapper/utils/getL1ToLongtailQuote.ts

```diff
@@ -0,0 +1,174 @@
+  const sellChainId = sellAsset.chainId
+  const buyChainId = buyAsset.chainId
+
+  const nativeSellAssetId = chainAdapterManager.get(sellChainId)?.getFeeAssetId()
+  const nativeSellAsset = nativeSellAssetId ? assetsById[nativeSellAssetId] : undefined
+
+  const nativeBuyAssetId = chainAdapterManager.get(buyChainId)?.getFeeAssetId()
+  const nativeBuyAsset = nativeBuyAssetId ? assetsById[nativeBuyAssetId] : undefined
+
+  if (!nativeBuyAsset) {
+    return Err(
+      makeSwapErrorRight({
+        message: `[getThorTradeQuote] - No native buy asset found for ${buyChainId}.`,
+        code: TradeQuoteError.InternalError,
+        details: { buyAssetChainId: buyChainId },
+      }),
+    )
+  }
+
+  if (!nativeSellAsset) {
+    return Err(
+      makeSwapErrorRight({
+        message: `[getThorTradeQuote] - No native buy asset found for ${sellChainId}.`,
+        code: TradeQuoteError.InternalError,
+        details: { sellAssetChainId: sellChainId },
+      }),
+    )
+  }
+
+  // TODO: Move these constants outside
+  // TODO2: use more than just UniswapV3, and also consider trianglar routes.
+  const ALLOWANCE_CONTRACT = '0xF892Fef9dA200d9E84c9b0647ecFF0F34633aBe8' // TSAggregatorTokenTransferProxy
+
+  const l1Tol1QuoteInput: GetTradeQuoteInput = {
+    ...input,
+    buyAsset: nativeBuyAsset,
+    sellAsset,
+    sellAmountIncludingProtocolFeesCryptoBaseUnit: sellAmountCryptoBaseUnit,
+  }
+
+  const thorchainQuotes = await getL1quote(
+    l1Tol1QuoteInput,
+    streamingInterval,
+    TradeType.L1ToLongTail,
+  )
+
+  const unwrappedThorchainQuotes = thorchainQuotes.unwrap()
+
+  let bestAggregator: AggregatorContract
+  let quotedAmountOut: bigint
+
+  const promises = await Promise.allSettled(
+    unwrappedThorchainQuotes.map(async quote => {
+      const onlyStep = quote.steps[0]
+
+      const maybeBestAggregator = await getBestAggregator(
+        nativeBuyAsset,
+        getWrappedToken(nativeBuyAsset),
+        getTokenFromAsset(buyAsset),
+        onlyStep.buyAmountAfterFeesCryptoBaseUnit,
+      )
+
+      if (maybeBestAggregator.isErr()) return Err(maybeBestAggregator.unwrapErr())
+
+      const unwrappedResult = maybeBestAggregator.unwrap()
+
+      bestAggregator = unwrappedResult.bestAggregator
+      quotedAmountOut = unwrappedResult.quotedAmountOut
+
+      const updatedMemo = addAggregatorAndDestinationToMemo({
+        aggregator: bestAggregator,
+        destinationToken: fromAssetId(buyAsset.assetId).assetReference as Address,
+        minAmountOut: convertDecimalPercentageToBasisPoints(
+          slippageTolerancePercentageDecimal ??
+            getDefaultSlippageDecimalPercentageForSwapper(SwapperName.Thorchain),
+        ).toString(),
+        quotedMemo: quote.memo,
+      })
+
+      const { data, router, vault } = await getEvmThorTxInfo({
+        sellAsset,
+        sellAmountCryptoBaseUnit,
+        memo: updatedMemo,
+        expiry: quote.expiry,
+      })
```

Comment:
We already call this in `getL1quote` under the hood, this is likely why I think the way I handled this isn't really elegant